patroni/postgresql.py

```diff
@@ -637,7 +651,18 @@ def sync_replication_slots(self, cluster):
         if self.use_slots:
             try:
                 self.load_replication_slots()
-                slots = [m.name for m in cluster.members if m.name != self.name] if self.role == 'master' else []
+                # if the replicatefrom tag is set on the member - we should not create the replication slot for it on
+                # the current master, because that member would replicate from elsewhere. We still create the slot if
+                # the replicatefrom destination member is currently not a member of the cluster (fallback to the
+                # master), or if replicatefrom destination member happens to be the current master
+                if self.role == 'master':
+                    slots = [m.name for m in cluster.members if m.name != self.name and
+                             (not cluster.has_member(m.replicatefrom)
+                              if m.replicatefrom and m.replicatefrom != self.name else True)]
```

Comment:
`m.replicatefrom is None or m.replicatefrom == self.name or not cluster.has_member(m.replicatefrom)`